oletools/msodde.py

```diff
@@ -992,50 +940,27 @@ def main(cmd_line_args=None):
     # Setup logging to the console:
     # here we use stdout instead of stderr by default, so that the output
     # can be redirected properly.
-    logging.basicConfig(level=LOG_LEVELS[args.loglevel], stream=sys.stdout,
-                        format='%(levelname)-8s %(message)s')
-    # enable logging in the modules:
-    log.setLevel(logging.NOTSET)
-
-    if args.json and args.loglevel.lower() == 'debug':
-        log.warning('Debug log output will not be json-compatible!')
+    log_helper.enable_logging(args.json, args.loglevel, stream=sys.stdout)
 
     if args.nounquote:
         global NO_QUOTES
         NO_QUOTES = True
 
-    if args.json:
-        jout = []
-        jout.append(BANNER_JSON)
-    else:
-        # print banner with version
-        print(BANNER)
-
-    if not args.json:
-        print('Opening file: %s' % args.filepath)
+    logger.log_at_current_level(BANNER)
```

Comment:
Why is the banner sent to the logger and not printed?

Reply:
We do that so we get valid json output. A printed banner would not be valid json and users might want to log to places other than stdout

Reply:
I think the banner should only be printed when the output is expected to be plain text. I don't think it belongs to logs, unless you need it in the JSON output. I have not used the JSON output much so far, so I need to test it in practice to understand how it works compared to plain text.

Reply:
@decalage2 
In order to avoid printing the banner when we have JSON enabled logging we would need to either add if-elses to every module that uses the logger or have the logger skip the banner when it's printing as JSON... at least those were the options I could think of. Since you mentioned that the banner shouldn't be handled by the logger, would you prefer the first option then?

Reply:
JSON example: https://www.json.org/example.html  Very convenient format because there are readers/writers for almost every programming language and it is simple to read for humans.

By "does not belong to logs" I guess you mean it should not appear in log files, right?
I suggest we create a function OletoolsLogger.print_banner() and call that at the start of each oletools main script. If the logger uses json it can implement/overwrite that function as a noop (or print to stderr?), while a "regular" (i.e. non-json) logger will just print it to stdout.